pandas/core/internals/managers.py

```diff
@@ -626,8 +631,12 @@ def comp(s, regex=False):
                 rb = new_rb
             result_blocks.extend(rb)
 
-        bm = type(self)(result_blocks, self.axes)
-        bm._consolidate_inplace()
+        if self.ndim == 1:
```

Comment:
i would make a function or method to do this, you seem to be repeating this code